src/controllers/auth.ts

```diff
@@ -69,3 +72,79 @@ export const login = async (req: Request, res: Response): Promise<void> => {
     token,
   })
 }
+export const updatePassword = async (req: Request, res: Response): Promise<void> => {
+  const { newPassword, token } = req.body
+  const decoded = jwt.verify(token, process.env.JWT_SECRET) as Payload
+  if (!decoded) {
+    res.status(StatusCodes.UNAUTHORIZED).json({ msg: 'Invalid credentials' })
+    return
+  }
+  const { tenantId } = decoded
+  const tenant = await Tenant.findOne({ _id: tenantId })
+  if (!tenant) {
+    res.status(StatusCodes.UNAUTHORIZED).json({ msg: 'Invalid credentials' })
+    return
+  }
+  if (newPassword.length < 6) {
+    res.status(StatusCodes.BAD_REQUEST).json({ msg: 'Password must be at least 6 characters' })
+    return
+  }
+  if (newPassword === tenant.password) {
+    res.status(StatusCodes.BAD_REQUEST).json({ msg: 'Please provide a different password' })
+    return
+  }
+  tenant.changedPasswords.push({
```

Comment:
ternant.password is undefined. I dont think we need the changed password property in the tenant database